src/cli/cli.cc

```diff
@@ -157,9 +160,59 @@ static CheckInResult checkIn(AkliteClientExt &client, CheckMode check_mode,
   }
 }
 
-StatusCode CheckIn(AkliteClientExt &client, const LocalUpdateSource *local_update_source, CheckMode check_mode) {
+static Json::Value getCheckInResultJson(AkliteClientExt &client, const CheckInResult &ci_res,
+                                        const GetTargetToInstallResult &gti_res) {
+  const auto current = client.GetCurrent();
+
+  Json::Value json_root;
+
+  auto app_shortlist = client.GetAppShortlist();
+  for (const auto &target : ci_res.Targets()) {
+    Json::Value json_target;
+    json_target["name"] = target.Name();
+    json_target["version"] = target.Version();
+    if (!gti_res.selected_target.IsUnknown() && gti_res.selected_target.Name() == target.Name()) {
+      json_target["selected"] = true;
+      json_target["reason"] = gti_res.reason;
+    }
+    if (client.IsRollback(target)) {
+      json_target["bad"] = true;
```

Comment:
Maybe we should call it `rollback`, so it looks like `"rollback": true`, which is more clear way to indicate that a rollback occurred when trying to install a given target? Just IMO.

Reply:
I find the "rollback" term a bit ambiguous. I even changed some variable names because of that during the recent refactoring. "rollback" may cause confusion as there are 2 targets involved during a rollback operation. "caused_rollback" could be more appropriate to indicate the bad one. "bad" should be clear enough, but I'm open to change it.

Reply:
Maybe "failing" then, or just "failed".

Reply:
`failed` sounds good.

Reply:
Changed to `failed`